docs/dev_manual_todo_examples.rb

```diff
@@ -0,0 +1,179 @@
+
+######################################
+# Update the database
+######################################
+
+# Update node Alice to add an age property
+result = driver.execute_query(
+  'MATCH (p:Person {name: $name})
+   SET p.age = $age',
+  nil,
+  { database: 'neo4j' },
+  name: 'Alice',
+  age: 42
+)
+
+summary = result.summary
+puts "Query updated the database?"
+puts summary.counters.contains_updates?
+
+# Create a relationship :KNOWS between Alice and Bob
+result = driver.execute_query(
+  'MATCH (alice:Person {name: $name})
+   MATCH (bob:Person {name: $friend})
+   CREATE (alice)-[:KNOWS]->(bob)',
+  nil,
+  { database: 'neo4j' },
+  name: 'Alice',
+  friend: 'Bob'
+)
+
+summary = result.summary
+puts 'Query updated the database?'
+puts summary.counters.contains_updates?
+
+######################################
+# Delete from the database
+######################################
+
+# Remove the Alice node and all its relationships
+result = driver.execute_query(
+  'MATCH (p:Person {name: $name})
+   DETACH DELETE p',
+  nil,
+  { database: 'neo4j' },
+  name: 'Alice'
+)
+
+summary = result.summary
+puts "Query updated the database?"
+puts summary.counters.contains_updates?
+
+######################################
+# Summary
+######################################
+
+# Retrieve the execution summary
+result = driver.execute_query(
+  "UNWIND ['Alice', 'Bob'] AS name
+   MERGE (p:Person {name: name})",
+  nil,
+  { database: 'neo4j' }
+)
+
+puts result.summary
+
+# Query counters
+result = driver.execute_query(
+  "MERGE (p:Person {name: $name})
+   MERGE (p)-[:KNOWS]->(:Person {name: $friend})",
+  nil,
+  { database: 'neo4j' },
+  name: 'Mark',
+  friend: 'Bob'
+)
+query_counters = result.summary.counters
+puts query_counters
+
+# Query execution plan
+result = driver.execute_query(
+  'EXPLAIN MATCH (p {name: $name}) RETURN p',
+  nil,
+  { database: 'neo4j' },
+  name: 'Alice'
+)
+query_plan = result.summary.plan.arguments['string-representation']
+puts query_plan
+
+# Notifications with Neo4j status codes
+result = driver.execute_query(
+  "MATCH p=shortestPath((:Person {name: $start})-[*]->(:Person {name: $end}))
+   RETURN p",
+  nil,
+  { database: 'neo4j' },
+  start: 'Alice',
+  end: 'Bob'
+)
+notifications = result.summary.notifications
+puts notifications
+
+# Notifications with GQL status codes
+result = execute_query(
+  "MATCH p=shortestPath((:Person {name: $start})-[*]->(:Person {name: $end}))
+   RETURN p",
+  nil,
+  { database: 'neo4j' },
+  start: 'Alice',
+  end: 'Bob'
+)
+statuses = result.summary.gql_status_objects
+puts statuses
+
+# Filter notifications
+# Allow only WARNING notifications, but not of HINT or GENERIC classifications
+driver = Neo4j::Driver::GraphDatabase.driver(
+  'bolt://localhost:7687',
+  Neo4j::Driver::AuthTokens.basic('neo4j', 'password'),
+  notification_config: {
+    minimum_severity: :warning, # use :off to disable entirely
+    disabled_categories: [:hint, :generic]
+  }
+)
```

Comment:
I think you should have at most one or two examples specifying the AuthToken and QueryConfig. Most examples should skip those parameters to show case the most common scenario (I believe 99%) of query and parameters only.

Reply:
Done